### PR TITLE
New library concept

### PR DESCRIPTION
**Pull request by SaschaMzH**
The library is moved into a subfolder to make it possible to simulate the I2C Lib from onion for a local debugging.
A base implementation for the local debugging from the onion Lib is also part of this PR.

### CODE REVIEW

**Comment by SaschaMzH** [comment]
[Local debugging](https://trello.com/c/P6AGX1Fg/30-local-debugging)

File: deploy.sh [hunk 1]
```diff
@@ -27,7 +27,12 @@ if [ \$1 ]; then
     path=\"\$1\"hucon
 fi
 
-mkdir -p \$path | tail -n+\$PAYLOAD_LINE \$0 | tar -xvzC \$path
+echo \"Remove old installation from \$path\"
+rm -rf \$path
```
Comment (JonasTrampe) [inline deploy.sh]: please do a check if the path exist before removal, otherwise the image generation will fail!

**Comment by JonasTrampe** [comment]
I am not sure if the image generation will work after this change. I have to check that.

File: deploy.sh (hunk 1 above)
Comment (SaschaMzH) [reply deploy.sh]: Check is implemented.

**Pull request merged**